pkg/config/event_listener.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright (c) 2018 Red Hat, Ine.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package config contains types and functions used to load the service configuration.
+//
+package config
+
+import (
+	"github.com/golang/glog"
+	"github.com/yaacov/observer/observer"
+)
+
+// ChangeEvent contains config change event info
+//
+type ChangeEvent struct {
+	// Empty
+}
+
+// ChangeListener a listener function that can be called when config event change is triggered.
+//
+type ChangeListener func(*ChangeEvent)
+
+// EventListener is an event listener for autoheal configuration object
+//
+type EventListener struct {
+	configFilesChangedObserver *observer.Observer
+	configFilesLoadedObserver  *observer.Observer
+}
+
+// Open the change obeserver channels
+//
+func (e *EventListener) Open() {
+	// Start a change watcher over changed config files.
+	if e.configFilesChangedObserver == nil {
+		e.configFilesChangedObserver = new(observer.Observer)
+		e.configFilesChangedObserver.Open()
+	}
+
+	// Start a change watcher over loaded config files.
+	if e.configFilesLoadedObserver == nil {
+		e.configFilesLoadedObserver = new(observer.Observer)
+		e.configFilesLoadedObserver.Open()
+	}
+}
+
+// ShutDown close the change obeserver channels
+//
+func (e *EventListener) ShutDown() {
```

Comment:
Can this method be private?

Reply:
we use this one in healer.go:

https://github.com/openshift/autoheal/pull/93/files#diff-3c6d47d7b38aeac558768f51b2845cadR168

Reply:
We use Config.ShutDown, if I understand correctly, not EventListener.ShutDown, right?

Reply:
Ahh... sorry, will fix :-)